pkg/dynamicaccess/history.go

```diff
@@ -1,36 +1,158 @@
 package dynamicaccess
 
 import (
-	"github.com/ethereum/go-ethereum/common"
-	"github.com/ethersphere/bee/v2/pkg/kvs"
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"strconv"
+	"time"
+
+	"github.com/ethersphere/bee/v2/pkg/file"
+	"github.com/ethersphere/bee/v2/pkg/manifest"
+	"github.com/ethersphere/bee/v2/pkg/manifest/mantaray"
+	"github.com/ethersphere/bee/v2/pkg/swarm"
 )
 
 type History interface {
-	Add(timestamp int64, kvs kvs.KeyValueStore) error
-	Get(timestamp int64) (kvs.KeyValueStore, error)
-	Lookup(at int64) (kvs.KeyValueStore, error)
+	Add(ctx context.Context, actRef swarm.Address, timestamp *int64) error
+	Lookup(ctx context.Context, timestamp int64, ls file.LoadSaver) (swarm.Address, error)
```

Comment:
loadsaver can be a member in history, then the same ls can be used as in NewHistory()

Reply:
fixed